tau2_integration/runners/saga_runner.py

```
@@ -0,0 +1,618 @@
+                        if tool_outputs:
+                            failure_context.append("Information gathered in this attempt:\n" + "\n".join(tool_outputs))
+                    else:
+                        logger.info("Replanning disabled. Stopping.")
+                        break
+                        
+                except Exception as e:
+                    logger.error(f"Iteration {iteration} failed: {e}")
+                    # Try last-ditch compensation
+                    try:
+                        self._compensate(tools, recorder)
+                    except:
```

Ruff (0.14.13)

Do not use bare `except`

(E722)